Library/Homebrew/extend/ENV.rb

```diff
@@ -4,11 +4,7 @@
 require "extend/ENV/super"
 
 def superenv?
-  if OS.linux?
-    ARGV.env == "super" || ARGV.homebrew_developer? && ARGV.env != "std"
```

Comment:
When `--env=std` is specified use `ENV/std` and ignore the value of `ARGV.homebrew_developer?`